src/Sarif.Viewer.VisualStudio.Interop/SarifViewerInterop.cs

```diff
@@ -97,36 +99,96 @@ public bool IsViewerExtensionLoaded
         /// Opens the specified SARIF log file in the SARIF Viewer extension.
         /// </summary>
         /// <param name="path">The path of the log file.</param>
-        public async Task<bool> OpenSarifLogAsync(string path)
+        public Task<bool> OpenSarifLogAsync(string path)
         {
             if (string.IsNullOrWhiteSpace(path))
             {
                 throw new ArgumentNullException(nameof(path));
             }
 
-            bool result = false;
+            return this.CallServiceApiAsync(ViewerLoadServiceInterfaceName, (service) =>
+            {
+                service.LoadSarifLogs(path);
+                return true;
+            });
+        }
 
-            if (IsViewerExtensionInstalled && (IsViewerExtensionLoaded || LoadViewerExtension() != null))
+        /// <summary>
+        /// Loads the specified SARIF logs in the viewer.
+        /// </summary>
+        /// <param name="paths">The complete path to the SARIF log files.</param>
+        public Task<bool> OpenSarifLogAsync(IEnumerable<string> paths)
+        {
+            return this.OpenSarifLogAsync(paths, promptOnLogConversions: true);
+        }
+
+        /// <summary>
+        /// Loads the specified SARIF logs in the viewer.
+        /// </summary>
+        /// <param name="paths">The complete path to the SARIF log files.</param>
+        /// <param name="promptOnLogConversions">Specifies whether the viewer should prompt if a SARIF log needs to be converted.</param>
+        /// <remarks>
+        /// Reasons for SARIF log file conversion include a conversion from a tool's log to SARIF, or a the SARIF schema version is not the latest version.
+        /// </remarks>
+        public Task<bool> OpenSarifLogAsync(IEnumerable<string> paths, bool promptOnLogConversions)
+        {
+            return this.CallServiceApiAsync(ViewerLoadServiceInterfaceName, (service) =>
             {
-                // Get the service interface type
-                Type[] types = ViewerExtensionAssembly.GetTypes();
-                Type sarifLoadServiceInterface = types.Where(t => t.Name == ViewerServiceInterfaceName).FirstOrDefault();
+                service.LoadSarifLogs(paths, promptOnLogConversions);
```

Comment:
>service [](start = 16, length = 7)

These helpers don't provide for an error path, return 'false'. is that right? what happens when this handler raises an unhandled exception? does VS hang? no apparent IDE impact?

Reply:
You are correct that the "Services" do not actually return any type of value. The original API was shipped this way so I did not want to introduce an public API change.

Since the service itself is internal and can be changed to return a value without changing the API surface, I'm am going to leave this code as is.

---
In reply to: [440858559](https://github.com/microsoft/sarif-visualstudio-extension/pull/143#discussion_r440858559) [](ancestors = 440858559)